msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/AbstractApplicationBase.java

```diff
@@ -0,0 +1,476 @@
+                CompletableFuture.runAsync(runnable, serviceBundle.getExecutorService()) :
+                CompletableFuture.runAsync(runnable);
+    }
+
+    CompletableFuture<IAuthenticationResult> executeRequest(
+            MsalRequest msalRequest) {
+
+        AuthenticationResultSupplier supplier = getAuthenticationResultSupplier(msalRequest);
+
+        ExecutorService executorService = serviceBundle.getExecutorService();
+        return executorService != null ?
+                CompletableFuture.supplyAsync(supplier, executorService) :
+                CompletableFuture.supplyAsync(supplier);
+
+    }
+
+    AuthenticationResult acquireTokenCommon(MsalRequest msalRequest, Authority requestAuthority)
+            throws Exception {
+
+        HttpHeaders headers = msalRequest.headers();
+
+        if (logPii) {
+            log.debug(LogHelper.createMessage(
+                    String.format("Using Client Http Headers: %s", headers),
+                    headers.getHeaderCorrelationIdValue()));
+        }
+
+        TokenRequestExecutor requestExecutor = new TokenRequestExecutor(
+                requestAuthority,
+                msalRequest,
+                serviceBundle);
+
+        AuthenticationResult result = requestExecutor.executeTokenRequest();
+
+        if (authenticationAuthority.authorityType.equals(AuthorityType.AAD)) {
+            InstanceDiscoveryMetadataEntry instanceDiscoveryMetadata =
+                    AadInstanceDiscoveryProvider.getMetadataEntry(
+                            requestAuthority.canonicalAuthorityUrl(),
+                            validateAuthority,
+                            msalRequest,
+                            serviceBundle);
+
+            tokenCache.saveTokens(requestExecutor, result, instanceDiscoveryMetadata.preferredCache);
+        } else {
+            tokenCache.saveTokens(requestExecutor, result, authenticationAuthority.host);
+        }
+
+        return result;
+    }
+
+    @Override
+    public CompletableFuture<IAuthenticationResult> acquireToken(RefreshTokenParameters parameters) {
+
+        validateNotNull("parameters", parameters);
+
+        RequestContext context = new RequestContext(
+                this,
+                PublicApi.ACQUIRE_TOKEN_BY_REFRESH_TOKEN,
+                parameters);
+
+        RefreshTokenRequest refreshTokenRequest = new RefreshTokenRequest(
+                parameters,
+                this,
+                context);
+
+        return executeRequest(refreshTokenRequest);
+    }
+
+    @Override
+    public CompletableFuture<IAuthenticationResult> acquireTokenSilently(SilentParameters parameters)
+            throws MalformedURLException {
+
+        validateNotNull("parameters", parameters);
+
+        RequestContext context;
+        if (parameters.account() != null) {
+            context = new RequestContext(
+                    this,
+                    PublicApi.ACQUIRE_TOKEN_SILENTLY,
```

Comment:
I don't think this should be part of the base.

Reply:
Moved back to AbstractClientApplicationBase, see this comment: https://github.com/AzureAD/microsoft-authentication-library-for-java/pull/715#discussion_r1346086456